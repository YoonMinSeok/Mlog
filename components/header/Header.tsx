import React from 'react';
import styled from 'styled-components';

import Link from 'next/link';

const Header = () => {
  return (
    <Container>
      <ContentContainer>
        <LogoContainer>
          <Link href="/">
            <Logo>Mlog</Logo>
          </Link>
        </LogoContainer>
        <LinkContainer>
          <Link href="/">
            <LinkMenu>Introduce</LinkMenu>
          </Link>
          <Link href="/">
            <LinkMenu>Posts</LinkMenu>
          </Link>
          <Link href="/">
            <LinkMenu>Plan</LinkMenu>
          </Link>
        </LinkContainer>
      </ContentContainer>
    </Container>
  );
};

export default Header;

const Container = styled.div`
  position: fixed;
  width: 200px;
  height: 100%;
  background: #1e1e1e;
`;

const ContentContainer = styled.div`
  dispaly: flex;
  padding: 25px 0 0 20px;
`;

const LogoContainer = styled.div`
  display: flex;
  width: 100%;
`;

const Logo = styled.a`
  font-size: 30px;
  color: #ffffff;
  font-family: fantasy;
`;

const LinkContainer = styled.div`
  display: flex;
  flex-flow: column;
  width: 70%;
  margin-top: 370px;
`;

const LinkMenu = styled.a`
  font-size: 25px;
  color: #ffffff;
  font-family: fantasy;
  margin: 20px 0;
  letter-spacing: 0;
  cursor: pointer;

  &:after {
    backface-visibility: hidden;
    border: 1px solid rgba(0, 0, 0, 0.5);
    bottom: 0px;
    content: ' ';
    display: block;
    margin: 0 auto;
    position: relative;
    transition: all 280ms ease-in-out;
    width: 0;
  }

  &:hover:after {
    backface-visibility: hidden;
    border-color: #fff;
    transition: width 350ms ease-in-out;
    width: 100%;
  }
`;
