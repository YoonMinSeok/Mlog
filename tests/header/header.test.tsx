//header test

import Header from '@components/header/Header';
import { render, screen, fireEvent } from '@testing-library/react';

describe('<Introduce />', () => {
  const setup = () => {
    render(<Header />);
    const firstLink = screen.getByText('Mlog');
    const secondLink = screen.getByText('Introduce');
    const thirdLink = screen.getByText('Posts');
    const fourthLink = screen.getByText('Plan');

    return {
      firstLink,
      secondLink,
      thirdLink,
      fourthLink,
    };
  };
  it('is correct link content', () => {
    const { firstLink, secondLink, thirdLink, fourthLink } = setup();
    expect(firstLink).toBeInTheDocument();
    expect(secondLink).toBeInTheDocument();
    expect(thirdLink).toBeInTheDocument();
    expect(fourthLink).toBeInTheDocument();
  });

  it('is true link', async () => {
    const { firstLink, secondLink, thirdLink, fourthLink } = setup();
    console.log(firstLink);
    const firstLinkTest = fireEvent.click(firstLink);
    console.log(firstLinkTest);
  });
});
