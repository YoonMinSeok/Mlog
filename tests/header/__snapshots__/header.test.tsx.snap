// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Introduce /> is true link 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bczRLJ jOdXOG"
      >
        <div
          class="sc-gsnTZi jyTlyf"
        >
          <div
            class="sc-dkzDqf hbSGLe"
          >
            <a
              class="sc-hKMtZM fZpMMJ"
              data-testid="Mlog"
            >
              Mlog
            </a>
          </div>
          <div
            class="sc-eCYdqJ bFOCbt"
          >
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Introduce"
            >
              Introduce
            </a>
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Posts"
            >
              Posts
            </a>
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Plan"
            >
              Plan
            </a>
          </div>
        </div>
      </header>
    </div>
    <div>
      <section
        class="sc-gKXOVf fstWrS"
      >
        <div
          class="sc-iBkjds iCLDcX"
        >
          <p>
            <span
              id="Introduce_Title1"
            >
              This page is MinSeok develop blog. 
            </span>
            <span
              id="Introduce_Title2"
            >
              I usally use Typescript, React, Redux.
            </span>
            <span
              id="Introduce_Title3"
            >
              Thanks for look
            </span>
          </p>
        </div>
        <div
          class="sc-ftvSup"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Introduce_Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="/img.jpg"
              srcset="/img.jpg 1x, /img.jpg 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-gKXOVf fstWrS"
    >
      <div
        class="sc-iBkjds iCLDcX"
      >
        <p>
          <span
            id="Introduce_Title1"
          >
            This page is MinSeok develop blog. 
          </span>
          <span
            id="Introduce_Title2"
          >
            I usally use Typescript, React, Redux.
          </span>
          <span
            id="Introduce_Title3"
          >
            Thanks for look
          </span>
        </p>
      </div>
      <div
        class="sc-ftvSup"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Introduce_Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="/img.jpg"
            srcset="/img.jpg 1x, /img.jpg 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Introduce /> is true link 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="sc-bczRLJ jOdXOG"
      >
        <div
          class="sc-gsnTZi jyTlyf"
        >
          <div
            class="sc-dkzDqf hbSGLe"
          >
            <a
              class="sc-hKMtZM fZpMMJ"
              data-testid="Mlog"
            >
              Mlog
            </a>
          </div>
          <div
            class="sc-eCYdqJ bFOCbt"
          >
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Introduce"
            >
              Introduce
            </a>
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Posts"
            >
              Posts
            </a>
            <a
              class="sc-jSMfEi ecuhYD"
              data-testid="Plan"
            >
              Plan
            </a>
          </div>
        </div>
      </header>
    </div>
    <div>
      <section
        class="sc-gKXOVf fstWrS"
      >
        <div
          class="sc-iBkjds iCLDcX"
        >
          <p>
            <span
              id="Introduce_Title1"
            >
              This page is MinSeok develop blog. 
            </span>
            <span
              id="Introduce_Title2"
            >
              I usally use Typescript, React, Redux.
            </span>
            <span
              id="Introduce_Title3"
            >
              Thanks for look
            </span>
          </p>
        </div>
        <div
          class="sc-ftvSup"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Introduce_Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="/img.jpg"
              srcset="/img.jpg 1x, /img.jpg 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
      </section>
    </div>
    <div>
      <section
        class="sc-gKXOVf fstWrS"
      >
        <div
          class="sc-iBkjds iCLDcX"
        >
          <p>
            <span
              id="Introduce_Title1"
            >
              This page is MinSeok develop blog. 
            </span>
            <span
              id="Introduce_Title2"
            >
              I usally use Typescript, React, Redux.
            </span>
            <span
              id="Introduce_Title3"
            >
              Thanks for look
            </span>
          </p>
        </div>
        <div
          class="sc-ftvSup"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
                style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
              />
            </span>
            <img
              alt="Introduce_Logo"
              data-nimg="intrinsic"
              decoding="async"
              src="/img.jpg"
              srcset="/img.jpg 1x, /img.jpg 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </span>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="sc-gKXOVf fstWrS"
    >
      <div
        class="sc-iBkjds iCLDcX"
      >
        <p>
          <span
            id="Introduce_Title1"
          >
            This page is MinSeok develop blog. 
          </span>
          <span
            id="Introduce_Title2"
          >
            I usally use Typescript, React, Redux.
          </span>
          <span
            id="Introduce_Title3"
          >
            Thanks for look
          </span>
        </p>
      </div>
      <div
        class="sc-ftvSup"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27400%27%20height=%27400%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="Introduce_Logo"
            data-nimg="intrinsic"
            decoding="async"
            src="/img.jpg"
            srcset="/img.jpg 1x, /img.jpg 2x"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </span>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
